metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuiteDiscoveryResult.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.meta.internal.metals.MetalsEnrichments._
+
+import org.eclipse.{lsp4j => l}
+
+final case class TestSuiteDiscoveryResult(
+    targetName: String,
+    targetUri: String,
+    discovered: java.util.List[TestSuiteDiscoveryResult.Discovered]
+)
+
+object TestSuiteDiscoveryResult {
+  sealed abstract trait Discovered {
+    def nonEmpty: Boolean = this match {
+      case p: Package => p.children.asScala.nonEmpty
+      case _: TestSuite => true
+    }
```

Comment:
Since we leave the hierarchy why not just implement the method in subclasses?
This look like a workaround for overloading.


```suggestion
    def nonEmpty: Boolean = true
```

Reply:
You're suggesting that pattern matching is worse than overloading? I think they're equal :D 
Anyway, it's thanks to your comment I improved what sealed hierarchy looks, now I don't have a `kind` field in subclasses which is great.